src/overlays/actors/ovl_En_Nb/z_en_nb.c

```diff
@@ -31,81 +122,669 @@ const ActorInit En_Nb_InitVars = {
+    s16 cutscene = func_80BC0050(this, 0);
+    s32 ret = false;
+
+    switch (this->behaviour) {
+        case ENNB_BEHAVIOUR_0:
+            if (!func_80BBFFD4(this, cutscene)) {
+                break;
+            }
+        // fallthrough
+        case ENNB_BEHAVIOUR_2:
+        case ENNB_BEHAVIOUR_4:
+        case ENNB_BEHAVIOUR_6:
+        case ENNB_BEHAVIOUR_8:
+            Camera_SetTargetActor(Play_GetCamera(play, ActorCutscene_GetCurrentSubCamId(cutscene)), &this->actor);
+            this->behaviour++;
+            ret = true;
+            break;
+
+        case ENNB_BEHAVIOUR_1:
+        case ENNB_BEHAVIOUR_3:
+        case ENNB_BEHAVIOUR_5:
+        case ENNB_BEHAVIOUR_7:
+            if ((this->actor.child != NULL) && (this->actor.child->update != NULL)) {
+                Camera_SetTargetActor(Play_GetCamera(play, ActorCutscene_GetCurrentSubCamId(cutscene)),
+                                      this->actor.child);
+            }
+            this->behaviour++;
+            ret = true;
+            break;
+
+        case ENNB_BEHAVIOUR_9:
+            ActorCutscene_Stop(cutscene);
+            this->behaviour++;
+            ret = true;
+            break;
+    }
+
+    return ret;
+}
+
+s32 func_80BC01DC(Actor* thisx, PlayState* play) {
+    EnNb* this = THIS;
+    s32 pad;
+    s32 ret = false;
+
+    switch (this->behaviour) {
+        case ENNB_BEHAVIOUR_0:
+            if (Player_GetMask(play) == PLAYER_MASK_ALL_NIGHT) {
+                this->behaviour = ENNB_BEHAVIOUR_1;
+            } else {
+                this->behaviour = ENNB_BEHAVIOUR_5;
+            }
+            break;
+
+        case ENNB_BEHAVIOUR_1:
+            func_8016A268(&play->state, 1, 0, 0, 0, 0);
+            this->unk_286 = 40;
+            this->behaviour = (u16)(this->behaviour + 1);
+            break;
+
+        case ENNB_BEHAVIOUR_2:
+            MREG(68) = (s16)(s32)(255.0f - (((f32)ABS_ALT(20 - this->unk_286) / 20.0f) * 255.0f));
+
+            if (this->unk_286 == 20) {
+                if (gSaveContext.eventInf[4] & 4) {
+                    play->interfaceCtx.unk_31B = 0;
+                } else {
+                    play->interfaceCtx.unk_31B = 1;
+                }
+                play->interfaceCtx.unk_31A = 6;
+                XREG(91) = 0xFF;
+            }
+
+            if (DECR(this->unk_286) == 0) {
+                this->behaviour++;
+            }
+            break;
+
+        case ENNB_BEHAVIOUR_3:
+            play->interfaceCtx.unk_31A = 4;
+            this->behaviour++;
+            ret = true;
+            break;
+
+        case ENNB_BEHAVIOUR_4:
+            play->interfaceCtx.unk_31A = 5;
+            this->behaviour++;
+            // fallthrough
+        case ENNB_BEHAVIOUR_5:
+            if (!(gSaveContext.eventInf[4] & 4)) {
+                gSaveContext.save.time = CLOCK_TIME(8, 0);
+                Sram_IncrementDay();
+            } else {
+                func_800FE658(120.0f);
+            }
```

Comment:
```suggestion
        case ENNB_BEHAVIOUR_1:
            // Setup a black fill-screen, although initialize to 0 alpha
            Play_FillScreen(&play->state, true, 0, 0, 0, 0);
            this->storyTimer = 40;
            this->behaviour = (u16)(this->behaviour + 1);
            break;

        case ENNB_BEHAVIOUR_2:
            // Slowly increase alpha to fill the screen with a black rectangle
            R_PLAY_FILL_SCREEN_ALPHA = (s16)(s32)(255.0f - (((f32)ABS_ALT(20 - this->storyTimer) / 20.0f) * 255.0f));

            if (this->storyTimer == 20) {
                if (gSaveContext.eventInf[4] & 4) {
                    play->interfaceCtx.storyType = STORY_TYPE_MASK_FESTIVAL;
                } else {
                    play->interfaceCtx.storyType = STORY_TYPE_GIANTS_LEAVING;
                }
                play->interfaceCtx.storyState = STORY_STATE_FADE_IN;
                R_STORY_FILL_SCREEN_ALPHA = 255;
            }

            if (DECR(this->storyTimer) == 0) {
                this->behaviour++;
            }
            break;

        case ENNB_BEHAVIOUR_3:
            play->interfaceCtx.storyState = STORY_STATE_SETUP_IDLE;
            this->behaviour++;
            ret = true;
            break;

        case ENNB_BEHAVIOUR_4:
            play->interfaceCtx.storyState = STORY_STATE_FADE_OUT;
            this->behaviour++;
            // fallthrough
        case ENNB_BEHAVIOUR_5:
            if (!(gSaveContext.eventInf[4] & 4)) {
                gSaveContext.save.time = CLOCK_TIME(8, 0);
                Sram_IncrementDay();
            } else {
                func_800FE658(120.0f);
            }
```
```suggestion
        case ENNB_BEHAVIOUR_1:
            func_8016A268(&play->state, 1, 0, 0, 0, 0);
            this->unk_286 = 40;
            this->behaviour = (u16)(this->behaviour + 1);
            break;

        case ENNB_BEHAVIOUR_2:
            MREG(68) = (s16)(s32)(255.0f - (((f32)ABS_ALT(20 - this->unk_286) / 20.0f) * 255.0f));

            if (this->unk_286 == 20) {
                if (gSaveContext.eventInf[4] & 4) {
                    play->interfaceCtx.unk_31B = 0;
                } else {
                    play->interfaceCtx.unk_31B = 1;
                }
                play->interfaceCtx.unk_31A = 6;
                XREG(91) = 0xFF;
            }

            if (DECR(this->unk_286) == 0) {
                this->behaviour++;
            }
            break;

        case ENNB_BEHAVIOUR_3:
            play->interfaceCtx.unk_31A = 4;
            this->behaviour++;
            ret = true;
            break;

        case ENNB_BEHAVIOUR_4:
            play->interfaceCtx.unk_31A = 5;
            this->behaviour++;
            // fallthrough
        case ENNB_BEHAVIOUR_5:
            if (!(gSaveContext.eventInf[4] & 4)) {
                gSaveContext.save.time = CLOCK_TIME(8, 0);
                Sram_IncrementDay();
            } else {
                func_800FE658(120.0f);
            }
```

Just want to make a note of story docs that tie into parameter/z_play.

Here's grandma's story update:
https://github.com/engineer124/mm/blob/21c3e39d22a8308bb7103244a17cead944bad764/src/code/z_parameter.c#L6826-L6858

Here's grandma's story draw:
https://github.com/engineer124/mm/blob/21c3e39d22a8308bb7103244a17cead944bad764/src/code/z_parameter.c#L5991-L6014

Here's grandma's story alloc/load:
https://github.com/engineer124/mm/blob/21c3e39d22a8308bb7103244a17cead944bad764/src/code/z_parameter.c#L6357-L6391